tools/flaky-test-reporter/main.go

```diff
@@ -64,9 +75,15 @@ func main() {
 		repoDataAll = append(repoDataAll, rd)
 	}
 
-	ghi, err := Setup(*githubToken)
-	if err != nil {
-		log.Fatalf("Cannot setup github: %v", err)
+	githubErr := ghi.processGithubIssues(repoDataAll, *dryrun)
```

Comment:
we should fail fast here and not create both clients if there is an error.
if githubErr {
log.Printf("gituhb errors: %v, err)
os.Exit(1)
}

Reply:
This was by design. There are 2 different kinds of errors might occur here, first one is failure that could cause inaccuracy, in this case it should fast fail, and this is handled by `fatalf` in github notification. The other kind is github operations failure, which doesn't affect Slack notification accuracy, so Slack notification should still be sent here.

Reply:
ok. thanks for explaining, Probably add a comment here then we need to verify both.

Reply:
added